source/common/http/conn_manager_impl.cc

```diff
@@ -1605,7 +1625,11 @@ void ConnectionManagerImpl::ActiveStream::onLocalReply(Code code) {
   // The BadRequest error code indicates there has been a messaging error.
   if (code == Http::Code::BadRequest && connection_manager_.codec_->protocol() < Protocol::Http2 &&
       !response_encoder_->streamErrorOnInvalidHttpMessage()) {
-    state_.saw_connection_close_ = true;
+    if (http1_connection_close_header_in_redirect_) {
```

Comment:
can we remove this boolean, always use the stream info version, and only copy from old to new in stream info's recreateStream iff the flag is set?
/wait

Reply:
Completely deprecated saw_connection_close_ in this PR. 
However, it worth to point out that there are 2 places the logic has been changed: recreateStream and decodeHeaders, and they are in different classes. Ideally I should have latched the runtime guard value in StreamInfoImpl and plumb it to both places, but that means added the bool in public interfaces. I didn't do so because the runtime guard being inconsistent in these 2 places looks fine to me.